app/services/return-requirements/submit-agreements-exceptions.service.js

```diff
@@ -0,0 +1,84 @@
+'use strict'
+
+/**
+ * Orchestrates validating the data for `/return-requirements/{sessionId}/agreements-exceptions` page
+ * @module SubmitAgreementsExceptions
+ */
+
+const AgreementsExceptionsPresenter = require('../../presenters/return-requirements/agreements-exceptions.presenter.js')
+const AgreementsExceptionsValidator = require('../../validators/return-requirements/agreements-exceptions.validator.js')
+const SessionModel = require('../../models/session.model.js')
+
+/**
+ * Orchestrates validating the data for `/return-requirements/{sessionId}/agreements-exceptions` page
+ *
+ * It first retrieves the session instance for the returns requirements journey in progress.
+ *
+ * The user input is then validated and the result is then combined with the output of the presenter to generate the
+ * page data needed by the view. If there was a validation error the controller will re-render the page so needs this
+ * information. If all is well the controller will redirect to the next page in the journey.
+ *
+ * @param {string} sessionId - The UUID of the current session
+ * @param {Object} payload - The submitted form data
+ *
+ * @returns {Promise<Object>} The page data for the agreements and exceptions page
+ */
+async function go (sessionId, payload) {
+  const session = await SessionModel.query().findById(sessionId)
+
+  _handleOneOptionSelected(payload)
+
+  const validationResult = _validate(payload)
+
+  if (!validationResult) {
+    await _save(session, payload)
+
+    return {}
+  }
+
+  const formattedData = AgreementsExceptionsPresenter.go(session, payload)
+
+  return {
+    activeNavBar: 'search',
+    error: validationResult,
+    pageTitle: 'Select agreements and exceptions for the requirements for returns',
+    ...formattedData
+  }
+}
+
+/**
+ * When a single agreement and exception is checked by the user, it returns as a string. When multiple agreements and exceptions are checked, the
+ * 'agreementsExceptions' is returned as an array. This function works to make those single selected string 'agreementsExceptions' into an array
+ * for uniformity.
+ */
```

Comment:
NIT

These lines are longer than 120 characters

```suggestion
/**
 * When a single agreement and exception is checked by the user, it returns as a string. When multiple agreements and
 * exceptions are checked the 'agreementsExceptions' is returned as an array. This function works to make those single
 * selected string 'agreementsExceptions' into an array for uniformity.
 */
```

I suggest you run the **🌈 Settings (WAL)** VSCode tasks in `wal-dev-environment` so you get some horizontal rules to indicate when you are getting close and when you've hit 120.

![Screenshot 2024-04-28 at 22 36 52](https://github.com/DEFRA/water-abstraction-system/assets/1789650/cc2ccbe4-b2a4-4a69-ba44-817d936699b8)